docs/write-lambdas-in-scala.md

```diff
@@ -1,9 +1,32 @@
-# Creating a Lambda with Scala <3
+# Creating a Lambda with Scala :see_no_evil:
 
-This is an almost step-by-step guide on how to run an AWS lambda using Scala. It is based on my experience
+An almost step-by-step guide on how to create an AWS lambda using Scala based on my experience
 writing the [Apple News Analytics Lambda](https://github.com/guardian/apple-news-analytics-lambda); you can use
 that repo as a reference.
 
+### Lambda uses cases
+
+You might want to consider using a lambda if:
+
+- Your app responds to (AWS) _events_ rather than to user actions: you want some actions to be triggered by an S3 event, a DyanmoDB write, you want to connect it to a Kinesis stream and so on.
+- You want to move your app from a poll logic to a push logic and have AWS handle it for you.
+- You want your app to do something on scheduled events.
+
+Check out more use cases [here](http://docs.aws.amazon.com/lambda/latest/dg/use-cases.html).
+
+Other goodies include:
+
+- You can test your lambda from the AWS web console by writing your test event and run it again the lambda.
+- Logs end up in Cloudwatch automatically.
+- Fast feedback cycle - just reupload your jar, no actual machine redeploy needed.
+- There's an official Java SDK that covers pretty much everything you could possibly need.
+
+### Lambda limitations:
+
+- Max timeout hardcoded to 5 minutes (deafult being 1 minute). It means that whatever your lambda does, it has to be done within 5 minutes top.
+- Unpredictable spin up overhead when running software on the JVM (ScalaJS anyone?).
+- No control whatsoever on the hardware (obviously). You can find a list of physical limitations [here](http://docs.aws.amazon.com/lambda/latest/dg/limits.html).
+
 ### Create a new Lambda
 
 1. First you need to package up your project. One easy way to do it is with [sbt assembly](https://github.com/sbt/sbt-assembly).
```

Comment:
The improvements and fixes are great, but it's still missing bullet point "0", a guide on creating a lambda with Scala. Given this is titled `Creating a Lambda with Scala` it feels like a critical ommision.

Would it be better named "a few thoughts on whether you should use a lambda and some guidelines on deploying the lambda after you've made it".